source/agora/test/Base.d

```diff
@@ -2300,8 +2304,10 @@ public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)
 
     auto all_configs = validator_configs.chain(full_node_configs).array;
 
-    immutable(Block)[] blocks = generateExtraBlocks(GenesisBlock,
-        test_conf.extra_blocks);
+    immutable(Block)[] blocks;
+    if (!test_conf.node.limit_test_validators ||
+        test_conf.node.limit_test_validators == GenesisBlock.header.enrollments.length)
+        blocks = generateExtraBlocks(GenesisBlock, test_conf.extra_blocks);
```

Comment:
Shouldn't we add `else` case or  `assert` in case of being unable to generate extra blocks? Also, it needs a comment about what the `if` condition checks.

Reply:
I'll update in the newer branch with PR https://github.com/bosagora/agora/pull/2892 which I created for just these changes.